### ISSUE TITLE
precision lower than mmdetection on coco2014?

### ISSUE THREAD

**Issue by zhangliukun** [facebookresearch/maskrcnn-benchmark#482]
## ❓ Questions and Help
I train the e2e_mask_rcnn_X_101_32x8d_FPN_1x.yaml as the given configs on coco2014，here is the final model:
```xml
maskrcnn_benchmark.trainer INFO: eta: 0:00:00  iter: 180000  loss: 0.5438 (0.6606)  
loss_box_reg: 0.0952 (0.1084)  loss_classifier: 0.2156 (0.2108)  loss_mask: 0.2093 (0.2437) 
loss_objectness: 0.0153 (0.0431)  loss_rpn_box_reg: 0.0171 (0.0547)  time: 0.6992 (0.6999)  
data: 0.0103 (0.0102)  lr: 0.000100  max mem: 5877
```

the evaluation result is as follows:
```xml
Evaluate annotation type *bbox*
DONE (t=39.88s).
Accumulating evaluation results...
DONE (t=6.57s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.361
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.573
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.392
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.215
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.395
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.460
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.308
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.493
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.519
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.333
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.557
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.647
Loading and preparing results...
DONE (t=4.56s)
creating index...
index created!
Running per image evaluation...
Evaluate annotation type *segm*
DONE (t=43.13s).
Accumulating evaluation results...
DONE (t=6.25s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.331
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.541
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.352
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.151
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.359
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.480
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.293
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.454
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.475
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.278
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.517
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.623
2019-02-22 08:10:21,759 maskrcnn_benchmark.inference INFO: OrderedDict([('bbox', OrderedDict([('AP', 0.36133887907093726), ('AP50', 0.5726409013708667), ('AP75', 0.39222788414397924), ('APs', 0.21467701327635888), ('APm', 0.39488310429312873), ('APl', 0.45967256947157153)])), ('segm', OrderedDict([('AP', 0.3311802775758692), ('AP50', 0.5414932401844765), ('AP75', 0.35199652963065065), ('APs', 0.15109910831968543), ('APm', 0.3589733553080113), ('APl', 0.4801035835717649)]))])
```

However，the result is much lower than the mmdetection results,which is as follows:
![image](https://user-images.githubusercontent.com/7829940/53215637-79e6c900-368c-11e9-98a3-4e66cf7702c4.png)

I wonder if you could help me to figure out why the precision is lower? thx

**Comment by fmassa** [comment]
Hi,

This is very weird. We have trained those models and we matched the results in Detectron, see https://github.com/facebookresearch/maskrcnn-benchmark/blob/master/MODEL_ZOO.md#end-to-end-faster-and-mask-r-cnn-baselines, which is 42.2 for boxes and  37.8 for masks

What is the version of PyTorch and `maskrcnn-benchmark` that you are using? For `maskrcnn-benchmark`, can you tell us what is the last commit in the codebase?

Also, what command did you use to train the model?

It would be good to understand what happened, but I don't think it was a regression that was added to maskrcnn-benchmark recently.

**Comment by zhangliukun** [comment]
@fmassa 
```json
- Pytorch :1.0.0   py3.6_cuda9.0.176_cudnn7.4.1_1
- maskrcnn-benchmark: 
branch master  
commit f8b0118392aae793771cfc6cd869ad0a601d5c92
Author: xelmirage <xelmirage@users.noreply.github.com>
Date:   Tue Feb 19 17:00:48 2019 +0800
- command 
export NGPUS=3
python -m torch.distributed.launch --nproc_per_node=$NGPUS
 /tools/train_net.py --config-file "configs/e2e_mask_rcnn_X_101_32x8d_FPN_1x.yaml"
- dataset
coco2014
```

The MAX_ITER is 180000 following the default configuration,so as other configurations.

I wonder if training for longer periods will lead to better results.

**Comment by fmassa** [comment]
@zhangliukun you are not following the right learning rate adaptation rules.
The default config is for 8 GPUs, you'd need to modify the number of iterations, the learning rates and the learning rate schedules in order to match the results from the config.
Check the information in https://github.com/facebookresearch/maskrcnn-benchmark#single-gpu-training for more details

If when modifying the number of iterations, learning rate and learning rate schedules as expected you still see lower accuracies, please let me know
